src/Swashbuckle.AspNetCore.Swagger/DependencyInjection/SwaggerBuilderExtensions.cs

```diff
@@ -21,9 +21,9 @@ public static class SwaggerBuilderExtensions
         public static IApplicationBuilder UseSwagger(this IApplicationBuilder app, SwaggerOptions options)
         {
 #if (!NETSTANDARD2_0)
-            return app.UseMiddleware<SwaggerMiddleware>(options, app.ApplicationServices.GetRequiredService<TemplateBinderFactory>());
+            return app.UseMiddleware<SwaggerMiddleware>(options, app.ApplicationServices.GetRequiredService<TemplateBinderFactory>(), app.ApplicationServices.GetService<ISwaggerDocumentSerializer>());
 #else
-            return app.UseMiddleware<SwaggerMiddleware>(options);
+            return app.UseMiddleware<SwaggerMiddleware>(options, app.ApplicationServices.GetService<ISwaggerDocumentSerializer>());
 #endif
```

Comment:
So looking at this code vs the constructor, this should be passing in `app.ApplicationServices` as the last parameter, rather than the first.

_However_ that is highlighting that the other change made recently (why you need to rebase) gets that out of the service provider _anyway_, so instead we could try simplifying this like so:

```csharp
public static IApplicationBuilder UseSwagger(this IApplicationBuilder app, SwaggerOptions options)
{
    return app.UseMiddleware<SwaggerMiddleware>(options, app.ApplicationServices);
}

public SwaggerMiddleware(
    RequestDelegate next,
    SwaggerOptions options,
    IServiceProvider serviceProvider)
{
    _next = next;
    _options = options ?? new SwaggerOptions();
    _requestMatcher = new TemplateMatcher(TemplateParser.Parse(_options.RouteTemplate), new RouteValueDictionary());
    _swaggerDocumentSerializer = serviceProvider?.GetService<ISwaggerDocumentSerializer>();
#if !NETSTANDARD
    _templateBinder = serviceProvider?.GetService<TemplateBinderFactory>()?.Create(RoutePatternFactory.Parse(_options.RouteTemplate));
#endif
}

public SwaggerMiddleware(RequestDelegate next, SwaggerOptions options)
    : this(next, options, null)
{
}
```

Reply:
Good point, done that now